db/models.py

```diff
@@ -0,0 +1,51 @@
+from sqlalchemy import Column, String, DateTime, ForeignKey
+from sqlalchemy.dialects.postgresql import UUID, JSONB
+from datetime import datetime
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship
+from pydantic import BaseModel, ConfigDict
+
+Base = declarative_base()
+
+
+class UserOrm(Base):
+    __tablename__ = "users"
+
+    id = Column(String, primary_key=True, unique=True, nullable=False)
+    name = Column(String, nullable=False)
+    email = Column(String, unique=True, nullable=False)
+    created_at = Column(DateTime, nullable=False, default=datetime.now())
+    updated_at = Column(DateTime, nullable=False, default=datetime.now())
+    threads = relationship("ThreadOrm", back_populates="user")
+
+
+class ThreadOrm(Base):
+    __tablename__ = "threads"
+
+    id = Column(String, primary_key=True, unique=True, nullable=False)
+    user_id = Column(String, ForeignKey("users.id"), nullable=False)
+    created_at = Column(DateTime, nullable=False, default=datetime.now())
+    updated_at = Column(
+        DateTime, nullable=False, default=datetime.now(), onupdate=datetime.now()
+    )
+    content = Column(JSONB, nullable=False)
```

Comment:
Is this intened to store the entire thread history? This could become pretty big, because it will contain all the output, including tools etc. Maybe we want a thread model and then a separate messages table where we split the individual messages into one row each. This would also make appending history easier.

Reply:
cc @kamicut how are you thinking about loading the thread data?

Reply:
There are a couple discussion threads in slack on this subject - but I agree that the conversations will become very big. A couple ideas: 
- We store conversation metadata in the postgres table, and then the actual conversation content in S3 
- We use a NoSQL database (DynamoDB, ElasticSearch) for the whole thing
- (as you suggested) We create a `messages` table where we store each message as a separate entry
- We store the conversation in the `threads` table but store any geometries separately, with references in the `threads.content` column 

Since there's still a lot of conversations happening around the Users<>Projects<>Threads structure I haven't pushed on this any more